pkg/reconciler/v1alpha1/taskrun/resources/input_resources.go

```diff
@@ -65,31 +75,70 @@ func AddInputResource(
 			return nil, fmt.Errorf("task %q failed to Get Pipeline Resource: %q", taskName, boundResource)
 		}
 
-		switch resource.Spec.Type {
-		case v1alpha1.PipelineResourceTypeGit:
-			{
-				gitResource, err = v1alpha1.NewGitResource(resource)
-				if err != nil {
-					return nil, fmt.Errorf("task %q invalid Pipeline Resource: %q", taskName, boundResource.ResourceRef.Name)
+		// if taskrun is fetching resource from previous task then execute copy step instead of fetching new copy
+		// to the desired destination directory
+		var copyStepsFromPrevTasks []corev1.Container
+
+		for i, path := range boundResource.Paths {
```

Comment:
Yeah I agree field name is not intuitive and it should be updated but refactoring this would add more changes to this PR. This is already growing too much in scope.  I want to follow up with another PR for that change

Reply:
yeah, makes sense since this was already there before this PR